File: src/Modules/Hermes.Modules.Books/Commands/Books/UpdateBook/UpdateBookCommand.cs
using Hermes.Domain.Enums;
using Hermes.Modules.Shared.Response;
using MediatR;

namespace Hermes.Modules.Books.Commands.Books.UpdateBook
{
    public sealed class UpdateBookCommand : IRequest<Response>
    {
        public int Id { get; set; } 
        public string Name { get; set; } = string.Empty;
        public int PublishYear { get; set; }
        public decimal Price { get; set; }
        public int Quantity { get; set; }
        public bool IsAvailable { get; set; }
        public byte[]? BookImage { get; set; }
        public int AuthorId { get; set; }
        public int PublisherId { get; set; }
        public BookCategory Category { get; set; } = BookCategory.None;
        public int AntiqueShopId { get; set; } = 1;
    }
}

File: src/Hermes.Web/src/types/books/UpdateBookCommand.ts
import BookCategory from "./BookCategory";

export default interface UpdateBookCommand {
    id: number;
    name: string;
    publishYear: number;
    price: number;
    quantity: number;
    isAvailable: boolean;
    bookImage?: string;
    authorId: number;
    authorName: string;
    authorSurname: string;
    publisherId: number;
    publisherName: string;
    category: BookCategory; 
    antiqueShopId: number;
}